MathLib/InterpolationAlgorithms/PiecewiseLinearInterpolation.cpp

```diff
@@ -95,7 +95,7 @@ double PiecewiseLinearInterpolation::getDerivative(
         interval_idx = 1;
     }
 
-    if (interval_idx > 2 && interval_idx < _supp_pnts.size() - 1)
+    if (interval_idx > 1 && interval_idx < _supp_pnts.size() - 2)
```

Comment:
I assume this change fixes some errors. Please extend the PiecewiseLinearInterpolation unit test to assure that this part is tested.